### PR TITLE
Minor: Add fetch from follower system test

### PR DESCRIPTION
**Pull request by mumrah**
This adds a basic system test that enables rack-aware brokers with the rack-aware replica selector for fetch from followers (KIP-392). The test asserts that the follower was read from at least once and that all the messages that were produced were successfully consumed.

### CODE REVIEW

**Comment by mumrah** [comment]
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2853/

**Comment by mumrah** [comment]
retest this please

File: tests/kafkatest/tests/core/fetch_from_follower_test.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        jmx_attribute = "preferred-read-replica"
+
+        jmx_object_name = "kafka.consumer:type=consumer-fetch-manager-metrics,client-id=%s,topic=%s,partition=%d" % \
+                  ("console-consumer", self.topic, 0)
+        self.jmx_tool.jmx_object_names = [jmx_object_name]
+        self.jmx_tool.jmx_attributes = [jmx_attribute]
+        self.jmx_tool.start_jmx_tool(consumer_idx, consumer_node)
+
+        # Wait for at least one interval of "metadata.max.age.ms"
+        time.sleep(5)
```
Comment (hachikuji) [inline tests/kafkatest/tests/core/fetch_from_follower_test.py]: nit: could we move the metadata max age to a variable or something?

File: tests/kafkatest/tests/core/fetch_from_follower_test.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                                        consumer_timeout_ms=60000, message_validator=is_int,
+                                        consumer_properties={"client.rack": non_leader_rack, "metadata.max.age.ms": "3000"})
+
+        # Start up and let some data get produced
+        self.start_producer_and_consumer()
+        time.sleep(5)
+
+        consumer_node = self.consumer.nodes[0]
+        consumer_idx = self.consumer.idx(consumer_node)
+        jmx_attribute = "preferred-read-replica"
```
Comment (hachikuji) [inline tests/kafkatest/tests/core/fetch_from_follower_test.py]: nit: maybe a clearer name is `read_replica_attribute` or something like that?

File: tests/kafkatest/tests/core/fetch_from_follower_test.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+        # Read the JMX output
+        self.jmx_tool.read_jmx_output(consumer_idx, consumer_node)
+
+        all_captured_preferred_read_replicas = defaultdict(int)
+        self.logger.debug(self.jmx_tool.jmx_stats)
+
+        for ts, data in self.jmx_tool.jmx_stats[0].items():
+            for k, v in data.items():
+                self.logger.debug("JMX => %s %s" % (k, v))
```
Comment (hachikuji) [inline tests/kafkatest/tests/core/fetch_from_follower_test.py]: Is this not too verbose?

File: tests/kafkatest/tests/core/fetch_from_follower_test.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                if k.endswith(jmx_attribute):
+                    all_captured_preferred_read_replicas[int(v)] += 1
+
+        self.logger.debug("Saw the following preferred read replicas %s",
+                          dict(all_captured_preferred_read_replicas.items()))
+        for idx in (1, 2, 3):
```
Comment (hachikuji) [inline tests/kafkatest/tests/core/fetch_from_follower_test.py]: This loop is a little puzzling to me. Is it not the same as the following:

```python
assert all_captured_preferred_read_replicas[non_leader_idx] > 0, ...
```

**Review by hachikuji — COMMENTED** [review]
Thanks, left a few comments.

File: tests/kafkatest/tests/core/fetch_from_follower_test.py (hunk 3 above)
Comment (mumrah) [reply tests/kafkatest/tests/core/fetch_from_follower_test.py]: It is -- this was left over from early testing/debugging

File: tests/kafkatest/tests/core/fetch_from_follower_test.py (hunk 4 above)
Comment (mumrah) [reply tests/kafkatest/tests/core/fetch_from_follower_test.py]: Heh, this is another hold over from the earlier version of this test -- i'll simplify

**Comment by mumrah** [comment]
@hachikuji thanks -- addressed your comments

**Comment by mumrah** [comment]
This is looking pretty stable https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2885/

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for the patch.

**Pull request merged**